docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+```toml
+initial_block_results_retain_height = 0
+```
+
+| Value type          | integer |
+|:--------------------|:--------|
+| **Possible values** | &gt;= 0 |
+
+### storage.pruning.data_companion.genesis_hash
+Hash of the Genesis file, passed to CometBFT via the command line.
+```toml
+genesis_hash = ""
+```
+
+| Value type          | string             |
+|:--------------------|:-------------------|
+| **Possible values** | hex-encoded number |
+|                     | `""`               |
+
+If this hash mismatches the hash that CometBFT computes on the genesis file, the node is not able to boot.
+
+## Transaction indexer
+Transaction indexer settings.
+
+The application will set which txs to index.
+In some cases, a node operator will be able to decide which txs to index based on the configuration set in the application.
+
+### tx_index.indexer
+What indexer to use for transactions.
+```toml
+indexer = "kv"
+```
+
+| Value type          | string   |
+|:--------------------|:---------|
+| **Possible values** | `"kv"`   |
+|                     | `"null"` |
+|                     | `"psql"` |
+
+`"null"` indexer indexes nothing.
+
+`"kv"` is the simplest possible indexer, backed by a key-value storage.
+The key-value storage database backend is defined in [`db_backend`](#db_backend).
+
+`"psql"` indexer is backed by an external PostgreSQL server.
+The server connection string is defined in [`tx_index.psql-conn`](#tx_indexpsql-conn).
+
+The transaction height and transaction hash is always indexed, except with the `"null"` indexer.
+
+### tx_index.psql-conn
+The PostgreSQL connection configuration.
+```toml
+psql-conn = ""
+```
+
+| Value type          | string                                                       |
+|:--------------------|:-------------------------------------------------------------|
+| **Possible values** | `"postgresql://<user>:<password>@<host>:<port>/<db>?<opts>"` |
+|                     | `""`                                                         |
+
+## Prometheus Instrumentation
+An extensive amount of Prometheus metrics are built into CometBFT.
+
+### instrumentation.prometheus
+Enable or disable presenting the Prometheus metrics at an endpoint.
+```toml
+prometheus = false
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `false` |
+|                     | `true`  |
+
+When enabled, metrics are served under the `/metrics` endpoint on the
+[instrumentation.prometheus_listen_addr](#instrumentationprometheus_listen_addr) address.
+
+### instrumentation.prometheus_listen_addr
+Address to listen for Prometheus collector(s) connections.
+```toml
+prometheus_listen_addr = ":26660"
+```
+
+| Value type          | string                                |
+|:--------------------|:--------------------------------------|
+| **Possible values** | Network address (`"127.0.0.1:26657"`) |
```

Comment:
It says "Network address" here, but its default value doesn't have an IP address